SurveyAppTests/Tests/Modules/SurveyDetail/SurveyDetailInteractorSpec.swift

```diff
@@ -27,6 +27,47 @@ final class SurveyDetailInteractorSpec: QuickSpec {
                 interactor = SurveyDetailInteractor(surveyService: surveyService)
                 interactor.output = output
             }
+            
+            describe("its fetchSurveyDetail is called") {
+                context("when the request returns success") {
+                    beforeEach {
+                        surveyService.fetchSurveyDetailIdCompletionClosure = { _, completion in
+                            let data: SurveyDetailResponse = JSON.SurveyService.fetchSurveyDetailSuccess.decoded()
+                            completion(.success(data))
+                            return nil
+                        }
+
+                        let surveyId = "d5de6a8f8f5f1cfe51bc"
+                        interactor.fetchSurveyDetail(id: surveyId)
+                    }
+
+                    it("triggers output to call didFetchSurveyDetail") {
+                        expect(output.didFetchSurveyDetailQuestionsCalled) == true
+                    }
+                    
+                    it("output should receive questions correctly") {
+                        expect(output.didFetchSurveyDetailQuestionsReceivedQuestions?.count) == 1
+                        expect(output.didFetchSurveyDetailQuestionsReceivedQuestions?.first?.text) == "Food â€“ Variety, Taste and Presentation"
+                        expect(output.didFetchSurveyDetailQuestionsReceivedQuestions?.first?.displayType) == .star
```

Comment:
please check these value with object from `let data: SurveyDetailResponse = JSON.SurveyService.fetchSurveyDetailSuccess.decoded()`

Reply:
@blyscuit The attributes in `SurveyDetailResponse` needs to cast to type `SurveyQuestionAttributes` in order to get `text` and `displayType`.

```Swift
override func spec() {
    ...

    let sampleSurveyDetail: SurveyDetailResponse = JSON.SurveyService.fetchSurveyDetailSuccess.decoded()
    let questionAttributes = sampleSurveyDetail.included[1].attributes as? SurveyQuestionAttributes

    ...
}
```
Would it be okay to implement this way? please help suggest 🙏